### DISCUSSION TITLE
Automatic calculation of the SSD size in FEMU/femu-scripts /run-blackbox.sh

### DISCUSSION THREAD

**Discussion by thesrsakabuvttchi** [MoatLab/FEMU#119] [Ideas]
In run-blackbox.sh, the parameter ssd_size, has a set formula based on the abovementioned parameters and can be calculated using them. I suggest tweaking this to calculate the ssd_size automatically. I have attached a sample code snippet below. 

```
secsz=512 # sector size in bytes
secs_per_pg=8 # number of sectors in a flash page
pgs_per_blk=256 # number of pages per flash block
blks_per_pl=256 # number of blocks per plane
pls_per_lun=1 # keep it at one, no multiplanes support
luns_per_ch=4 # number of chips per channel
nchs=1 #number of channels
ssd_size=$(echo "scale=25; ($secsz*$secs_per_pg*$pgs_per_blk*$blks_per_pl*$pls_per_lun*$luns_per_ch*$nchs*0.75)/(1024*1024)"|bc)

```

This snippet returns the ssd_size as a floating point number, we can make changes to return an integer as well.

**Comment by huaicheng** [comment]
Yes, you're right. The maintainer was too lazy :).
